tools/testing/test_selections.py

```diff
@@ -91,9 +91,13 @@ def calculate_shards(
     tests: List[str],
     test_file_times: Dict[str, float],
     must_serial: Optional[Callable[[str], bool]] = None,
+    debug: bool = False
 ) -> List[Tuple[float, List[ShardedTest]]]:
     must_serial = must_serial or (lambda x: True)
 
+    if debug:
```

Comment:
Let's print all the inputs including `num_shards`, `tests`, `test_file_times`, and `must_serial`.  We can then try to reproduce the issue locally with them I think